docs/library/machine.Pin.rst

```diff
@@ -3,16 +3,42 @@
 class Pin -- control I/O pins
 =============================
 
-A pin is the basic object to control I/O pins (also known as GPIO -
-general-purpose input/output). It has methods to set
-the mode of the pin (input, output, etc) and methods to get and set the
-digital logic level. For analog control of a pin, see the ADC class.
+A pin object is used to control I/O pins (also known as GPIO - general-purpose
+input/output).  Pin objects are commonly associated with a physical pin that can
+drive a voltage and read voltages.  The pin class has methods to set the mode of
+the pin (IN, OUT, etc) and methods to get and set the digital logic level.
+For analog control of a pin, see the ADC class.
 
-Usage Model:
+A specific pin is constructed by using its identifier, which is specific to a
+given port.  Identifiers can always be integers and on some ports strings can be
```

Comment:
Back when @danicampora and I were thinking about the new HW API, and drafting the wiki, we thought it would be good to *require* all ports to allow integers as identifiers in all the peripheral constructors.  The reason being that, coming to any new port, you always knew that you could construct a peripheral using an integer (and didn't need to read the docs to see how that particular port specified its ids).

We can discuss whether this is indeed a good idea or not.  But if you look at the wiki as it stands now, under "General conventions of the API", you'll see: "A port must allow to specify ids by a sequential integer starting at 0 and going to MAX".